test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -830,40 +843,68 @@ class test_ucp_peer_failure_rndv_abort : public test_ucp_peer_failure {
 
 test_ucp_peer_failure_rndv_abort *test_ucp_peer_failure_rndv_abort::self = nullptr;
 
-UCS_TEST_SKIP_COND_P(test_ucp_peer_failure_rndv_abort, get_zcopy,
-                     !is_proto_enabled(), "RNDV_THRESH=0",
-                     "RNDV_SCHEME=get_zcopy")
+UCS_TEST_P(test_ucp_peer_failure_rndv_abort, get_zcopy, "RNDV_SCHEME=get_zcopy")
 {
     rndv_progress_failure_test(rndv_mode::rndv_get, true);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_peer_failure_rndv_abort, put_zcopy_force_flush,
-                     !is_proto_enabled(), "RNDV_THRESH=0",
-                     "RNDV_SCHEME=put_zcopy", "RNDV_PUT_FORCE_FLUSH=y")
+UCS_TEST_P(test_ucp_peer_failure_rndv_abort, put_zcopy_force_flush,
+           "RNDV_SCHEME=put_zcopy", "RNDV_PUT_FORCE_FLUSH=y")
 {
     rndv_progress_failure_test(rndv_mode::rndv_put, true);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_peer_failure_rndv_abort,
-                     get_zcopy_memory_invalidation, !is_proto_enabled(),
-                     "RNDV_THRESH=0", "RNDV_SCHEME=get_zcopy")
+UCS_TEST_P(test_ucp_peer_failure_rndv_abort, get_zcopy_memory_invalidation,
+           "RNDV_SCHEME=get_zcopy")
 {
     rndv_progress_failure_test(rndv_mode::rndv_get, false);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_peer_failure_rndv_abort,
-                     put_zcopy_memory_invalidation, !is_proto_enabled(),
-                     "RNDV_THRESH=0", "RNDV_SCHEME=put_zcopy")
+UCS_TEST_P(test_ucp_peer_failure_rndv_abort, put_zcopy_memory_invalidation,
+           "RNDV_SCHEME=put_zcopy")
 {
     rndv_progress_failure_test(rndv_mode::rndv_put, false);
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_peer_failure_rndv_abort,
-                     put_zcopy_force_flush_memory_invalidation,
-                     !is_proto_enabled(), "RNDV_THRESH=0",
-                     "RNDV_SCHEME=put_zcopy", "RNDV_PUT_FORCE_FLUSH=y")
+UCS_TEST_P(test_ucp_peer_failure_rndv_abort,
+           put_zcopy_force_flush_memory_invalidation, "RNDV_SCHEME=put_zcopy",
+           "RNDV_PUT_FORCE_FLUSH=y")
 {
     rndv_progress_failure_test(rndv_mode::rndv_put, false);
 }
 
 UCP_INSTANTIATE_TEST_CASE(test_ucp_peer_failure_rndv_abort)
+
+
+class test_ucp_peer_failure_rndv_put_ppln_abort :
+      public test_ucp_peer_failure_rndv_abort {
+public:
+    static void get_test_variants(variant_vec_t &variants)
+    {
+        if (!mem_buffer::is_gpu_supported()) {
+            return;
+        }
+
+        test_ucp_peer_failure_rndv_abort::get_test_variants(variants);
+    }
+
+    void init() override
+    {
+        modify_config("RNDV_SCHEME", "put_ppln");
+        modify_config("RNDV_PIPELINE_SHM_ENABLE", "n");
+        m_mem_type = UCS_MEMORY_TYPE_CUDA;
```

Comment:
why only cuda?

Reply:
it's just to test a bug in abort flow, do you suggest to also add `cuda-managed`?